Issue 14: Docker image prepare (#14)

ravibeta opened the pull request:
**Change log description**  
Prepares the binding jars in the images so that they can be deployed on containers in K8s or Docker.
Fixes #14

**Purpose of the change**  
Fixes #14

**What the code does**  
It prepares and pushes the docker image so that they can be included as additional storage adapters. 

**How to verify it**  
- point the registry/repository to private
- run ./gradlew docker and ./gradlew dockerPush

Commit:
Issue 14: Docker image prepare (#14)
    Build steps for docker and dockerPush.
    Dockerfiles with jar addition

Commit:
Issue 14: Docker image prepare (#14)
    Build steps for docker and dockerPush.
    Dockerfiles with jar addition

Commit:
specify source and destination

Signed-off-by: Ravi Rajamani <ravi.rajamani@emc.com>

sachin-j-joshi reviewed (approved):
Approving to unblock testing

Commit:
Merge branch 'feature-separate-adapters' into issue-14-docker-push

Commit:
add back publishAllJars

Signed-off-by: Ravi Rajamani <ravi.rajamani@emc.com>

Commit:
remove unused

Signed-off-by: Ravi Rajamani <ravi.rajamani@emc.com>

ravibeta commented on gradle/maven-publish.gradle:
```diff
@@ -0,0 +1,104 @@
+
+            // Declares coordinates for publishing
+            groupId = 'io.pravega'
+            artifactId = "pravega-bindings-" + project.name.replace(':', '-')
```
no rename

sachin-j-joshi reviewed (commented):
1. Use the built jar as is without renaming it.
2. From the original image, remove original bindings jar.

sachin-j-joshi reviewed (changes requested):
Use the built jar as is without renaming it.
From the original image, remove original bindings jar.

Commit:
support removal of earlier bindings by testing it
Testing:
docker build --build-arg DOCKER_IMAGE_REGISTRY="devops-repo.isus.emc.com:8116/rajamr2" --build-arg DOCKER_IMAGE_REPOSITORY=pravega --build-arg DOCKER_IMAGE_VERSION="0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin" --build-arg PRAVEGA_BINDINGS_VERSION="0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT" --no-cache --tag devops-repo.isus.emc.com:8116/rajamr2/pravega:0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT -f docker/pravega/Dockerfile .
Sending build context to Docker daemon  3.655MB
Step 1/15 : ARG DOCKER_IMAGE_REGISTRY
 --->
Step 2/15 : ARG DOCKER_IMAGE_REPOSITORY
 --->
Step 3/15 : ARG DOCKER_IMAGE_VERSION
 --->
Step 4/15 : ARG PRAVEGA_IMAGE="${DOCKER_IMAGE_REGISTRY}/${DOCKER_IMAGE_REPOSITORY}:${DOCKER_IMAGE_VERSION}"
 --->
Step 5/15 : FROM ${PRAVEGA_IMAGE}
 ---> 1598ee51f4c2
Step 6/15 : WORKDIR /opt/pravega
 ---> 1536ab6e97bb
Removing intermediate container ab7e531996c4
Step 7/15 : ARG PRAVEGA_BINDINGS_VERSION
 ---> Running in 55ecb3e65209
 ---> 5260d4b1e48c
Removing intermediate container 55ecb3e65209
Step 8/15 : ARG HDFS_JAR="hdfs/build/libs/pravega-hdfs-${PRAVEGA_BINDINGS_VERSION}.jar"
 ---> Running in a78a7399d29b
 ---> b928f21fa8f1
Removing intermediate container a78a7399d29b
Step 9/15 : ARG S3_JAR="extendeds3/build/libs/pravega-extendeds3-${PRAVEGA_BINDINGS_VERSION}.jar"
 ---> Running in b0b478c443fd
 ---> c4db5759b205
Removing intermediate container b0b478c443fd
Step 10/15 : COPY ${HDFS_JAR} /opt/pravega/lib
 ---> 8db01bcbc000
Step 11/15 : COPY ${S3_JAR} /opt/pravega/lib
 ---> f6d5b8f54dfc
Step 12/15 : RUN rm -rf /opt/pravega/lib/pravega-bindings-*
 ---> Running in 35547a29b1c6
 ---> 91bfbbbd5537
Removing intermediate container 35547a29b1c6
Step 13/15 : RUN ls -1 /opt/pravega/lib | sort
 ---> Running in e0955ab7a810
FastInfoset-1.2.15.jar
HdrHistogram-2.1.11.jar
LatencyUtils-2.0.3.jar
accessors-smart-1.2.jar
animal-sniffer-annotations-1.17.jar
annotations-4.1.1.4.jar
aopalliance-1.0.jar
aopalliance-repackaged-2.5.0.jar
apacheds-i18n-2.0.0-M15.jar
apacheds-kerberos-codec-2.0.0-M15.jar
api-asn1-api-1.0.0-M20.jar
api-util-1.0.0-M20.jar
asm-3.1.jar
asm-5.0.4.jar
audience-annotations-0.5.0.jar
avro-1.7.4.jar
bookkeeper-common-4.9.2.jar
bookkeeper-common-allocator-4.9.2.jar
bookkeeper-proto-4.9.2.jar
bookkeeper-server-4.9.2.jar
bookkeeper-stats-api-4.9.2.jar
bookkeeper-tools-framework-4.9.2.jar
cglib-2.2.1-v20090111.jar
checker-qual-2.10.0.jar
circe-checksum-4.9.2.nar
commons-cli-1.2.jar
commons-codec-1.10.jar
commons-collections-3.2.2.jar
commons-collections4-4.1.jar
commons-compress-1.20.jar
commons-configuration-1.10.jar
commons-daemon-1.0.13.jar
commons-io-2.6.jar
commons-lang-2.6.jar
commons-lang3-3.7.jar
commons-logging-1.2.jar
commons-math3-3.1.1.jar
commons-net-3.1.jar
cpu-affinity-4.9.2.nar
curator-client-4.0.1.jar
curator-framework-4.0.1.jar
curator-recipes-4.0.1.jar
curator-test-4.0.1.jar
error_prone_annotations-2.3.4.jar
failureaccess-1.0.1.jar
google-auth-library-credentials-0.13.0.jar
grizzly-framework-2.4.4.jar
grizzly-http-2.4.4.jar
grizzly-http-server-2.4.4.jar
grpc-api-1.24.2.jar
grpc-auth-1.24.2.jar
grpc-context-1.24.2.jar
grpc-core-1.24.2.jar
grpc-netty-1.24.2.jar
grpc-protobuf-1.24.2.jar
grpc-protobuf-lite-1.24.2.jar
grpc-stub-1.24.2.jar
gson-2.7.jar
guava-28.2-jre.jar
guice-3.0.jar
guice-servlet-3.0.jar
hadoop-annotations-2.8.5.jar
hadoop-auth-2.8.5.jar
hadoop-common-2.8.5-tests.jar
hadoop-common-2.8.5.jar
hadoop-hdfs-2.8.5-tests.jar
hadoop-hdfs-2.8.5.jar
hadoop-hdfs-client-2.8.5.jar
hadoop-mapreduce-client-app-2.8.5.jar
hadoop-mapreduce-client-common-2.8.5.jar
hadoop-mapreduce-client-core-2.8.5.jar
hadoop-mapreduce-client-hs-2.8.5.jar
hadoop-mapreduce-client-jobclient-2.8.5-tests.jar
hadoop-mapreduce-client-jobclient-2.8.5.jar
hadoop-mapreduce-client-shuffle-2.8.5.jar
hadoop-minicluster-2.8.5.jar
hadoop-yarn-api-2.8.5.jar
hadoop-yarn-client-2.8.5.jar
hadoop-yarn-common-2.8.5.jar
hadoop-yarn-server-applicationhistoryservice-2.8.5.jar
hadoop-yarn-server-common-2.8.5.jar
hadoop-yarn-server-nodemanager-2.8.5.jar
hadoop-yarn-server-resourcemanager-2.8.5.jar
hadoop-yarn-server-tests-2.8.5-tests.jar
hadoop-yarn-server-web-proxy-2.8.5.jar
hk2-api-2.5.0.jar
hk2-locator-2.5.0.jar
hk2-utils-2.5.0.jar
htrace-core4-4.0.1-incubating.jar
http-server-4.9.2.jar
httpclient-4.5.2.jar
httpcore-4.4.4.jar
istack-commons-runtime-3.0.7.jar
j2objc-annotations-1.3.jar
jackson-annotations-2.9.10.jar
jackson-core-2.9.10.jar
jackson-core-asl-1.9.13.jar
jackson-databind-2.9.10.3.jar
jackson-dataformat-yaml-2.9.8.jar
jackson-jaxrs-1.8.3.jar
jackson-jaxrs-1.9.13.jar
jackson-mapper-asl-1.9.13.jar
jackson-module-jaxb-annotations-2.9.9.jar
jackson-xc-1.8.3.jar
jackson-xc-1.9.13.jar
jakarta.annotation-api-1.3.4.jar
jakarta.inject-2.5.0.jar
jakarta.ws.rs-api-2.1.5.jar
java-xmlbuilder-0.4.jar
javassist-3.22.0-CR2.jar
javax.activation-api-1.2.0.jar
javax.inject-1.jar
javax.inject-2.5.0-b32.jar
javax.servlet-api-4.0.0.jar
javax.ws.rs-api-2.0.1.jar
javax.ws.rs-api-2.1.jar
jaxb-api-2.3.1.jar
jaxb-impl-2.2.3-1.jar
jaxb-runtime-2.3.1.jar
jcip-annotations-1.0-1.jar
jcommander-1.48.jar
jctools-core-2.1.2.jar
jdom2-2.0.6.jar
jersey-apache-client4-1.19.3.jar
jersey-client-1.19.3.jar
jersey-client-2.29.jar
jersey-common-2.29.jar
jersey-container-grizzly2-http-2.29.jar
jersey-container-servlet-core-2.25.1.jar
jersey-core-1.19.3.jar
jersey-entity-filtering-2.29.jar
jersey-guice-1.9.jar
jersey-hk2-2.29.jar
jersey-json-1.9.jar
jersey-media-jaxb-2.29.jar
jersey-media-json-jackson-2.29.jar
jersey-media-multipart-2.25.1.jar
jersey-server-1.9.jar
jersey-server-2.29.jar
jets3t-0.9.0.jar
jettison-1.1.jar
jetty-6.1.26.jar
jetty-sslengine-6.1.26.jar
jetty-util-6.1.26.jar
jjwt-0.9.1.jar
jna-3.2.7.jar
jsch-0.1.54.jar
json-smart-2.3.jar
jsp-api-2.1.jar
jsr305-3.0.2.jar
leveldbjni-all-1.8.jar
listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
log4j-1.2.17.jar
logback-classic-1.2.3.jar
logback-core-1.2.3.jar
lzma-sdk-4j-9.22.0.jar
micrometer-core-1.2.0.jar
micrometer-registry-influx-1.2.0.jar
micrometer-registry-statsd-1.2.0.jar
mimepull-1.9.6.jar
netty-buffer-4.1.50.Final.jar
netty-codec-4.1.50.Final.jar
netty-codec-http-4.1.50.Final.jar
netty-codec-http2-4.1.50.Final.jar
netty-codec-socks-4.1.50.Final.jar
netty-common-4.1.50.Final.jar
netty-handler-4.1.50.Final.jar
netty-handler-proxy-4.1.50.Final.jar
netty-resolver-4.1.50.Final.jar
netty-tcnative-boringssl-static-2.0.30.Final.jar
netty-transport-4.1.50.Final.jar
netty-transport-native-epoll-4.1.50.Final-linux-x86_64.jar
netty-transport-native-epoll-4.1.50.Final.jar
netty-transport-native-unix-common-4.1.50.Final.jar
nimbus-jose-jwt-4.41.1.jar
object-client-3.0.6.jar
object-transform-1.1.0.jar
okhttp-2.4.0.jar
okio-1.4.0.jar
opencensus-api-0.21.0.jar
opencensus-contrib-grpc-metrics-0.21.0.jar
osgi-resource-locator-1.0.3.jar
paranamer-2.3.jar
perfmark-api-0.19.0.jar
pravega-client-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-common-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-controller-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-controller-pathing-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-extendeds3-0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT.jar
pravega-hdfs-0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT.jar
pravega-segmentstore-contracts-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-server-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-server-host-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-server-host-pathing-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-storage-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-storage-impl-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-authplugin-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-cluster-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-controller-api-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-metrics-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-protocol-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-security-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-standalone-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-standalone-pathing-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
proto-google-common-protos-1.12.0.jar
protobuf-java-3.5.1.jar
reflections-0.9.11.jar
rocksdbjni-5.13.1.jar
servlet-api-2.5-20081211.jar
servlet-api-2.5.jar
slf4j-api-1.7.25.jar
smart-client-2.1.1.jar
snakeyaml-1.23.jar
snappy-java-1.0.4.1.jar
stax-ex-1.8.jar
swagger-annotations-1.5.22.jar
swagger-core-1.5.22.jar
swagger-jaxrs-1.5.22.jar
swagger-jersey2-jaxrs-1.5.22.jar
swagger-models-1.5.22.jar
txw2-2.3.1.jar
validation-api-2.0.1.Final.jar
xercesImpl-2.9.1.jar
xmlenc-0.52.jar
zookeeper-3.5.5-tests.jar
zookeeper-3.5.5.jar
zookeeper-jute-3.5.5.jar
 ---> 7994306cbdeb
Removing intermediate container e0955ab7a810
Step 14/15 : RUN chmod +x -R /opt/pravega/lib/
 ---> Running in 2c607f870205
 ---> 2c242a1298ad
Removing intermediate container 2c607f870205
Step 15/15 : ENTRYPOINT /opt/pravega/scripts/entrypoint.sh
 ---> Running in 4c3adf26777e
 ---> a9924eef2f14
Removing intermediate container 4c3adf26777e
Successfully built a9924eef2f14
Successfully tagged devops-repo.isus.emc.com:8116/rajamr2/pravega:0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT

Signed-off-by: Ravi Rajamani <ravi.rajamani@emc.com>

ravibeta commented:
Yes Sachin I did

Commit:
add option
Testing:
 # docker build --build-arg DOCKER_IMAGE_REGISTRY="devops-repo.isus.emc.com:8116/rajamr2" --build-arg DOCKER_IMAGE_REPOSITORY=pravega --build-arg DOCKER_IMAGE_VERSION="0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin" --build-arg PRAVEGA_BINDINGS_VERSION="0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT" --no-cache --tag devops-repo.isus.emc.com:8116/rajamr2/pravega:0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT -f docker/pravega/Dockerfile .
Sending build context to Docker daemon  3.757MB
Step 1/16 : ARG DOCKER_IMAGE_REGISTRY
 --->
Step 2/16 : ARG DOCKER_IMAGE_REPOSITORY
 --->
Step 3/16 : ARG DOCKER_IMAGE_VERSION
 --->
Step 4/16 : ARG PRAVEGA_IMAGE="${DOCKER_IMAGE_REGISTRY}/${DOCKER_IMAGE_REPOSITORY}:${DOCKER_IMAGE_VERSION}"
 --->
Step 5/16 : FROM ${PRAVEGA_IMAGE}
 ---> 1598ee51f4c2
Step 6/16 : WORKDIR /opt/pravega
 ---> 6d4aaad32cc6
Removing intermediate container 06dce16293e0
Step 7/16 : ARG PRAVEGA_BINDINGS_VERSION
 ---> Running in 4d16355287ef
 ---> c3510b813dc0
Removing intermediate container 4d16355287ef
Step 8/16 : ARG HDFS_JAR="hdfs/build/libs/pravega-hdfs-${PRAVEGA_BINDINGS_VERSION}.jar"
 ---> Running in 78c204c1c5f8
 ---> 6dbd5aa76401
Removing intermediate container 78c204c1c5f8
Step 9/16 : ARG S3_JAR="extendeds3/build/libs/pravega-extendeds3-${PRAVEGA_BINDINGS_VERSION}.jar"
 ---> Running in 1586102bd607
 ---> bfe78a1e665b
Removing intermediate container 1586102bd607
Step 10/16 : COPY ${HDFS_JAR} /opt/pravega/lib
 ---> fe831a2b5d39
Step 11/16 : COPY ${S3_JAR} /opt/pravega/lib
 ---> 1ce1f2acb6f1
Step 12/16 : ARG ONLY
 ---> Running in 7ef1bc0d931d
 ---> 04c0308d2f0f
Removing intermediate container 7ef1bc0d931d
Step 13/16 : RUN if [ -z "$ONLY" ]; then echo "keeping earlier bindings jar"; else rm -rf /opt/pravega/lib/pravega-bindings-*; fi
 ---> Running in 3a10acaf717d
keeping earlier bindings jar
 ---> bed6b913d7e5
Removing intermediate container 3a10acaf717d
Step 14/16 : RUN ls -1 /opt/pravega/lib | sort
 ---> Running in fcb29c5c4489
FastInfoset-1.2.15.jar
HdrHistogram-2.1.11.jar
LatencyUtils-2.0.3.jar
accessors-smart-1.2.jar
animal-sniffer-annotations-1.17.jar
annotations-4.1.1.4.jar
aopalliance-1.0.jar
aopalliance-repackaged-2.5.0.jar
apacheds-i18n-2.0.0-M15.jar
apacheds-kerberos-codec-2.0.0-M15.jar
api-asn1-api-1.0.0-M20.jar
api-util-1.0.0-M20.jar
asm-3.1.jar
asm-5.0.4.jar
audience-annotations-0.5.0.jar
avro-1.7.4.jar
bookkeeper-common-4.9.2.jar
bookkeeper-common-allocator-4.9.2.jar
bookkeeper-proto-4.9.2.jar
bookkeeper-server-4.9.2.jar
bookkeeper-stats-api-4.9.2.jar
bookkeeper-tools-framework-4.9.2.jar
cglib-2.2.1-v20090111.jar
checker-qual-2.10.0.jar
circe-checksum-4.9.2.nar
commons-cli-1.2.jar
commons-codec-1.10.jar
commons-collections-3.2.2.jar
commons-collections4-4.1.jar
commons-compress-1.20.jar
commons-configuration-1.10.jar
commons-daemon-1.0.13.jar
commons-io-2.6.jar
commons-lang-2.6.jar
commons-lang3-3.7.jar
commons-logging-1.2.jar
commons-math3-3.1.1.jar
commons-net-3.1.jar
cpu-affinity-4.9.2.nar
curator-client-4.0.1.jar
curator-framework-4.0.1.jar
curator-recipes-4.0.1.jar
curator-test-4.0.1.jar
error_prone_annotations-2.3.4.jar
failureaccess-1.0.1.jar
google-auth-library-credentials-0.13.0.jar
grizzly-framework-2.4.4.jar
grizzly-http-2.4.4.jar
grizzly-http-server-2.4.4.jar
grpc-api-1.24.2.jar
grpc-auth-1.24.2.jar
grpc-context-1.24.2.jar
grpc-core-1.24.2.jar
grpc-netty-1.24.2.jar
grpc-protobuf-1.24.2.jar
grpc-protobuf-lite-1.24.2.jar
grpc-stub-1.24.2.jar
gson-2.7.jar
guava-28.2-jre.jar
guice-3.0.jar
guice-servlet-3.0.jar
hadoop-annotations-2.8.5.jar
hadoop-auth-2.8.5.jar
hadoop-common-2.8.5-tests.jar
hadoop-common-2.8.5.jar
hadoop-hdfs-2.8.5-tests.jar
hadoop-hdfs-2.8.5.jar
hadoop-hdfs-client-2.8.5.jar
hadoop-mapreduce-client-app-2.8.5.jar
hadoop-mapreduce-client-common-2.8.5.jar
hadoop-mapreduce-client-core-2.8.5.jar
hadoop-mapreduce-client-hs-2.8.5.jar
hadoop-mapreduce-client-jobclient-2.8.5-tests.jar
hadoop-mapreduce-client-jobclient-2.8.5.jar
hadoop-mapreduce-client-shuffle-2.8.5.jar
hadoop-minicluster-2.8.5.jar
hadoop-yarn-api-2.8.5.jar
hadoop-yarn-client-2.8.5.jar
hadoop-yarn-common-2.8.5.jar
hadoop-yarn-server-applicationhistoryservice-2.8.5.jar
hadoop-yarn-server-common-2.8.5.jar
hadoop-yarn-server-nodemanager-2.8.5.jar
hadoop-yarn-server-resourcemanager-2.8.5.jar
hadoop-yarn-server-tests-2.8.5-tests.jar
hadoop-yarn-server-web-proxy-2.8.5.jar
hk2-api-2.5.0.jar
hk2-locator-2.5.0.jar
hk2-utils-2.5.0.jar
htrace-core4-4.0.1-incubating.jar
http-server-4.9.2.jar
httpclient-4.5.2.jar
httpcore-4.4.4.jar
istack-commons-runtime-3.0.7.jar
j2objc-annotations-1.3.jar
jackson-annotations-2.9.10.jar
jackson-core-2.9.10.jar
jackson-core-asl-1.9.13.jar
jackson-databind-2.9.10.3.jar
jackson-dataformat-yaml-2.9.8.jar
jackson-jaxrs-1.8.3.jar
jackson-jaxrs-1.9.13.jar
jackson-mapper-asl-1.9.13.jar
jackson-module-jaxb-annotations-2.9.9.jar
jackson-xc-1.8.3.jar
jackson-xc-1.9.13.jar
jakarta.annotation-api-1.3.4.jar
jakarta.inject-2.5.0.jar
jakarta.ws.rs-api-2.1.5.jar
java-xmlbuilder-0.4.jar
javassist-3.22.0-CR2.jar
javax.activation-api-1.2.0.jar
javax.inject-1.jar
javax.inject-2.5.0-b32.jar
javax.servlet-api-4.0.0.jar
javax.ws.rs-api-2.0.1.jar
javax.ws.rs-api-2.1.jar
jaxb-api-2.3.1.jar
jaxb-impl-2.2.3-1.jar
jaxb-runtime-2.3.1.jar
jcip-annotations-1.0-1.jar
jcommander-1.48.jar
jctools-core-2.1.2.jar
jdom2-2.0.6.jar
jersey-apache-client4-1.19.3.jar
jersey-client-1.19.3.jar
jersey-client-2.29.jar
jersey-common-2.29.jar
jersey-container-grizzly2-http-2.29.jar
jersey-container-servlet-core-2.25.1.jar
jersey-core-1.19.3.jar
jersey-entity-filtering-2.29.jar
jersey-guice-1.9.jar
jersey-hk2-2.29.jar
jersey-json-1.9.jar
jersey-media-jaxb-2.29.jar
jersey-media-json-jackson-2.29.jar
jersey-media-multipart-2.25.1.jar
jersey-server-1.9.jar
jersey-server-2.29.jar
jets3t-0.9.0.jar
jettison-1.1.jar
jetty-6.1.26.jar
jetty-sslengine-6.1.26.jar
jetty-util-6.1.26.jar
jjwt-0.9.1.jar
jna-3.2.7.jar
jsch-0.1.54.jar
json-smart-2.3.jar
jsp-api-2.1.jar
jsr305-3.0.2.jar
leveldbjni-all-1.8.jar
listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
log4j-1.2.17.jar
logback-classic-1.2.3.jar
logback-core-1.2.3.jar
lzma-sdk-4j-9.22.0.jar
micrometer-core-1.2.0.jar
micrometer-registry-influx-1.2.0.jar
micrometer-registry-statsd-1.2.0.jar
mimepull-1.9.6.jar
netty-buffer-4.1.50.Final.jar
netty-codec-4.1.50.Final.jar
netty-codec-http-4.1.50.Final.jar
netty-codec-http2-4.1.50.Final.jar
netty-codec-socks-4.1.50.Final.jar
netty-common-4.1.50.Final.jar
netty-handler-4.1.50.Final.jar
netty-handler-proxy-4.1.50.Final.jar
netty-resolver-4.1.50.Final.jar
netty-tcnative-boringssl-static-2.0.30.Final.jar
netty-transport-4.1.50.Final.jar
netty-transport-native-epoll-4.1.50.Final-linux-x86_64.jar
netty-transport-native-epoll-4.1.50.Final.jar
netty-transport-native-unix-common-4.1.50.Final.jar
nimbus-jose-jwt-4.41.1.jar
object-client-3.0.6.jar
object-transform-1.1.0.jar
okhttp-2.4.0.jar
okio-1.4.0.jar
opencensus-api-0.21.0.jar
opencensus-contrib-grpc-metrics-0.21.0.jar
osgi-resource-locator-1.0.3.jar
paranamer-2.3.jar
perfmark-api-0.19.0.jar
pravega-bindings-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-client-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-common-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-controller-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-controller-pathing-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-extendeds3-0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT.jar
pravega-hdfs-0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT.jar
pravega-segmentstore-contracts-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-server-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-server-host-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-server-host-pathing-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-storage-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-segmentstore-storage-impl-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-authplugin-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-cluster-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-controller-api-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-metrics-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-protocol-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-shared-security-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-standalone-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
pravega-standalone-pathing-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin.jar
proto-google-common-protos-1.12.0.jar
protobuf-java-3.5.1.jar
reflections-0.9.11.jar
rocksdbjni-5.13.1.jar
servlet-api-2.5-20081211.jar
servlet-api-2.5.jar
slf4j-api-1.7.25.jar
smart-client-2.1.1.jar
snakeyaml-1.23.jar
snappy-java-1.0.4.1.jar
stax-ex-1.8.jar
swagger-annotations-1.5.22.jar
swagger-core-1.5.22.jar
swagger-jaxrs-1.5.22.jar
swagger-jersey2-jaxrs-1.5.22.jar
swagger-models-1.5.22.jar
txw2-2.3.1.jar
validation-api-2.0.1.Final.jar
xercesImpl-2.9.1.jar
xmlenc-0.52.jar
zookeeper-3.5.5-tests.jar
zookeeper-3.5.5.jar
zookeeper-jute-3.5.5.jar
 ---> 8d8c24eb3721
Removing intermediate container fcb29c5c4489
Step 15/16 : RUN chmod +x -R /opt/pravega/lib/
 ---> Running in ee7ccdcb365d
 ---> a620c7aacfd0
Removing intermediate container ee7ccdcb365d
Step 16/16 : ENTRYPOINT /opt/pravega/scripts/entrypoint.sh
 ---> Running in aefd4ea6512c
 ---> 06882ddaea1f
Removing intermediate container aefd4ea6512c
Successfully built 06882ddaea1f
Successfully tagged devops-repo.isus.emc.com:8116/rajamr2/pravega:0.9.0-2625.fd290ad-12.92a8547-SNAPSHOT

Signed-off-by: Ravi Rajamani <ravi.rajamani@emc.com>

Commit:
trim

Signed-off-by: Ravi Rajamani <ravi.rajamani@emc.com>

sachin-j-joshi commented on docker/pravega/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+WORKDIR /opt/pravega
+ARG PRAVEGA_BINDINGS_VERSION
+ARG HDFS_JAR="hdfs/build/libs/pravega-hdfs-${PRAVEGA_BINDINGS_VERSION}.jar"
+ARG S3_JAR="extendeds3/build/libs/pravega-extendeds3-${PRAVEGA_BINDINGS_VERSION}.jar"
```
Where are we removing old bindings jar ?

ravibeta replied:
Yes Sachin, it was there. Let me check. Thanks,

ravibeta replied:
https://github.com/pravega/pravega-storage-adapters/pull/15/files#diff-ec695cab72c3c30c1f15b1bcd0dafbfcR24

ravibeta replied:
Line 24

ravibeta commented on docker/pravega/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+COPY ${HDFS_JAR} /opt/pravega/lib
+COPY ${S3_JAR} /opt/pravega/lib
+ARG ONLY
+RUN if [ -z "$ONLY" ]; then echo "keeping earlier bindings jar"; else rm -rf /opt/pravega/lib/pravega-bindings-*; fi
```
This is where we remove earlier bindings jar. Thanks

sachin-j-joshi reviewed (approved):
Looks okay

ravibeta commented:
Yes Sachin Thank you very much. May I please ask you to merge this approved pr ? Thanks, Ravi
